### PR TITLE
Sign in connection #10 Tony/Spencer

### PR DESCRIPTION
**Pull request by tony1759**
## Describe your changes
Added user signup functionality with MongoDB integration:
- Created POST /api/users endpoint for user creation
- Added MongoDB User schema with validation
- Implemented duplicate email/username checking
- Connected frontend signup form to backend API
- Added proper error handling and user feedback
- Fixed form input field names to match backend expectations

## Issue ticket number and link
#10 

## Does your code meet the acceptance criteria?
- [x] Yes
  - ✓ Successfully creates new users in MongoDB database
  - ✓ Prevents duplicate email registration with error message
  - ✓ Prevents duplicate username registration with error message
  - ✓ Validates password matching
  - ✓ Properly handles and displays error messages

## Testing Done
1. Created new user successfully:
   - Username: test1
   - Email: test1@test.com
   - Result: User created in MongoDB

2. Tested duplicate prevention:
   - Attempted duplicate email - received error
   - Attempted duplicate username - received error

3. Tested form validation:
   - Password mismatch detection works
   - Required field validation works
## Other comments
- MongoDB connection and schema validation working as expected
- Error handling implemented for both frontend and backend
- User feedback implemented through alerts
- Form fields properly connected to backend API

Screenshot 2024-10-25 at 10 39 01 PM

Screenshot 2024-10-25 at 10 34 31 PM

Screenshot 2024-10-25 at 10 45 00 PM

Screenshot 2024-10-25 at 10 45 44 PM

### CODE REVIEW

**Comment by myix765** [comment]
Hi @tony1759 and @Sanderson28801, really nice job on completing the ticket! One change I would suggest is instead of hardcoding the backend URL to fetch from, could you instead set it so that the port is retrieved from a .env file instead? That way if people have different ports set up we don't get an error from accessing the server using the wrong URL. Let me know if you have any questions!

File: src/pages/SignUp.jsx [hunk 1]
```diff
@@ -8,17 +10,34 @@ export default function SignUp() {
...
-  const handleSubmit = (e) => {
+  const handleSubmit = async (e) => {
     e.preventDefault();
-    const { username, email, password, retypedPassword } = formData
+    const { username, email, password, retypedPassword} = formData
     if (password != retypedPassword) {
       alert(`Passwords do not match:\npassword: ${password}\nretyped password: ${retypedPassword}`)
     } else {
+      try {
+        const response = await axios.post('http://localhost:5000/api/users', formData);
```
Comment (myix765) [inline src/pages/SignUp.jsx]: Please change the URL here to retrieve the port from a .env file

**Review by myix765 — COMMENTED** [review]
After checking into your branch and testing if I can create a user, I'm getting an "An error occurred while creating the user." I tested using Postman and was able to succesfully create a user, but the user was created in the wrong database. 
Postman:
image
MongoDB:
Screenshot 2024-10-26 at 5 23 29 PM

Could you test the code with our database? I made a database for our project that everyone should be invited to, let me know if you don't see the invite!

Another thing is the sign up form doesn't seem to be setting the username field in formData (when submitting the form the username is blank in the alert). This actually isn't a problem with your code, it's a problem in main too, but if you could get that working that would be great (the issue is that the name attribute in the username input does not match the formData field)

**Comment by tony1759** [comment]
Screenshot 2024-10-26 at 8 32 17 PM
Screenshot 2024-10-26 at 8 32 56 PM
Screenshot 2024-10-26 at 8 34 42 PM

**Review by myix765 — APPROVED** [review]

**Pull request merged**